-01 version initial update

FrankXL opened the pull request:
Hi, this is the initial update of the 01 version, please review and give your comments on it.

FrankXL committed:
-01 version initial update

Hi, this is the initial update of the 01 version, please review and give your comments on it.

muhammad-usama-sardar commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -178,8 +203,6 @@ TBD
 # Security Considerations
 
-Risk of relay attacks needs to be evaluated in the design.
-
```
Since it is removed, what prevents a relay attack in this design?

FrankXL replied:
Usama, could you first write a section on the threat analysis of relay attacks?

muhammad-usama-sardar replied:
Sure, but I need some time until the end of next week. I think cutoff is 21st, so we can make it.

muhammad-usama-sardar commented:
I think the draft should clarify the comparison with [Trustee](https://github.com/confidential-containers/trustee).

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -77,6 +77,7 @@ For example, to protect model parameters from leakage and tampering, model weigh
 At this point, there are multiple options for the secure channel protocol that can be established, such as TLS, IPSec, QUIC, OHTTP, etc.
 The user may only need to complete E2E key negotiation based on remote attestation.
 As for which secure protocol or application layer encryption the negotiated key is used for, and how it is used, there can be various implementation methods.
+* Scenario 3: A company intends to deploy its private large model or file system to a public cloud platform. However, for security reasons, it first uploads the encrypted objects to the cloud platform. Subsequently, it needs to conduct a security check on the TEE (Trusted Execution Environment) within the cloud platform's computing environment using remote attestation. Once the security check is  ...
```
need a reference to diversion attack, such as
https://datatracker.ietf.org/meeting/interim-2025-rats-01/materials/slides-interim-2025-rats-01-sessa-identity-crisis-in-attested-tls-for-confidential-computing-00.pdf

and the corresponding defintion.

muhammad-usama-sardar replied:
I can add some text here. Early next week I hope to release a preprint. We can directly reference that then, rather than referencing a presentation.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -91,41 +92,37 @@ In summary, considering the diversity of remote attestation application scenario
+            |  Attester  |4. Attestation  |    Party    | Result against
+            |            |   Result +     |    & KMS    | appraisal policy
+            '------------' Raw public key '-------------' + Distribute Keys
+                           or Certificate
```
in step 3,5 raw public key or certificate needed to be signed

FrankXL replied:
In fact, in steps 1 and 3, the Attester's identity (raw public key or certificate) each requires signing by the Attester and Verifier's RATS signing keys. I will revise the latest diagram and add textual explanations.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -134,32 +131,60 @@ The KMS can generate or distribute these keys accordingly. During key rotation,
+          |  & Server  | 6.Attestation |    Party    | Result against
+          |            | Result        |    & Client | appraisal policy
+          '------------' + pubS        '-------------' + SK = privC * pubS
+        2. SK = privS * pubC
```
in step 5,6 raw public key or certificate needed to be signed

FrankXL replied:
In fact, in steps 3 and 5, the Attester's identity (raw public key or certificate) each requires signing by the Attester and Verifier's RATS signing keys. I will revise the latest diagram and add textual explanations.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -134,32 +131,60 @@ The KMS can generate or distribute these keys accordingly. During key rotation,
+            |            |   Result +     |      KMS    | appraisal policy
+            '------------' Raw public key '-------------' + Distribute Keys
+                           or Certificate
+
```
in step 3,4 raw public key or certificate needed to be signed

FrankXL replied:
In fact, in steps 1 and 3, the Attester's identity (raw public key or certificate) each requires signing by the Attester and Verifier's RATS signing keys. I will revise the latest diagram and add textual explanations.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -178,8 +203,6 @@ TBD
 # Security Considerations
 
-Risk of relay attacks needs to be evaluated in the design.
-
```
The underlying conveyance protocol should provide the protections (like data signature, secure channel, etc) between Attester, Verifier and Relying Party to prevent relay attack.

muhammad-usama-sardar replied:
Conveyance protocol alone cannot prevent relay attack. There has to be binding too.

junzhanghw replied:
In addition, certain channel binding protocols should be used to establish the connection between the Attester, Verifier, and Relying Party.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -91,41 +92,37 @@ In summary, considering the diversity of remote attestation application scenario
+            |  Attester  |4. Attestation  |    Party    | Result against
+            |            |   Result +     |    & KMS    | appraisal policy
+            '------------' Raw public key '-------------' + Distribute Keys
+                           or Certificate
```
in step 1,3, to use hash of raw public key or certificate, then in step 4, the information because AR+signed hash+raw pub key/certificate

FrankXL replied:
What do you mean by signed hash here? What is its purpose?

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -134,32 +131,60 @@ The KMS can generate or distribute these keys accordingly. During key rotation,
+          |  & Server  | 6.Attestation |    Party    | Result against
+          |            | Result        |    & Client | appraisal policy
+          '------------' + pubS        '-------------' + SK = privC * pubS
+        2. SK = privS * pubC
```
in step 3,5, to use hash of raw public key or certificate, then in step 6, the information because AR+signed hash+raw pubS

FrankXL replied:
What do you mean by signed hash here? What is its purpose?

junzhanghw replied:
I mean that using the hash(raw public key) reduces the payload to be delivered from Attester to Verifier, and Verifier to the Attester, as actually the Verifier does not appraise it and just sign it. So finally the Attester can send the raw public key, and the Attestation Result to the Relying Party, while the hash of raw public key is inside the signed Attestation Result.

So there are two possible options

1. Raw Public key/Cerificate
2. hash(raw public key/certificate)

also, in the option two, when the raw pub key/certificate remains unchanged for the reconnection with the relying party, there is even no need to send the raw public/key from attester to the relying party, as long as the relying party has a cache of the raw public key as it can compare the hash inside the attestation result with the hash of raw public key in its local data set.

FrankXL replied:
Understood, using the hash of the raw public key/certificate can be an optimized option, but I'm not sure if this optimization is necessary. What do others think?
In any case, we can add a description of this option in the text section. Could you update it directly?

junzhanghw replied:
I have changed my mind a little bit. So for the first time connection between the Attester and Relying Party, it is better to use the raw public key / certificate directly as you can embed them into the Attestation Result. For the second time/third time while the raw public key/certificate remains unchanged, then you can use the hash of them, as the Relying Party has already the cache of the raw public key/certificate, and it can compare the received hash, with the hash of pub key/certificate in its cache.

FrankXL replied:
> I have changed my mind a little bit. So for the first time connection between the Attester and Relying Party, it is better to use the raw public key / certificate directly as you can embed them into the Attestation Result. For the second time/third time while the raw public key/certificate remains unchanged, then you can use the hash of them, as the Relying Party has already the cache of the raw public key/certificate, and it can compare the received hash, with the hash of pub key/certificate in its cache.

yes, this is the right way. Could you add some text about it?

junzhanghw replied:
I have created one new section “Optimization Considerations” and add some text into it. I have launched the Pull Request for this change.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -134,32 +131,60 @@ The KMS can generate or distribute these keys accordingly. During key rotation,
+            |            |   Result +     |      KMS    | appraisal policy
+            '------------' Raw public key '-------------' + Distribute Keys
+                           or Certificate
+
```
in step 1,3, to use hash of raw public key or certificate, then in step 4, the information because AR+signed hash+raw pub  key/certificate

FrankXL replied:
What do you mean by signed hash here? What is its purpose?

FrankXL committed:
Update draft-xia-rats-key-negotiation-integration.md

2025-10-10 update, Detailed descriptions of the binding mechanism have been added to each section, along with revised diagrams and other modifications.

FrankXL committed:
Update draft-xia-rats-key-negotiation-integration.md

nits change.

FrankXL committed:
Update draft-xia-rats-key-negotiation-integration.md

solve the github error: contains trailing whitespace error.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -77,7 +77,9 @@ For example, to protect model parameters from leakage and tampering, model weigh
-* Scenario 3: A company intends to deploy its private large model or file system to a public cloud platform. However, for security reasons, it first uploads the encrypted objects to the cloud platform. Subsequently, it needs to conduct a security check on the TEE (Trusted Execution Environment) within the cloud platform's computing environment using remote attestation. Once the security check is  ...
+* Scenario 3: A company intends to deploy its private large model or file system to a public cloud platform. However, for security reasons, it first uploads the encrypted objects to the cloud platform. Subsequently, it needs to conduct a security check on the TEE (Trusted Execution Environment) within the cloud platform's computing environment using remote attestation. Once the security check is  ...
+
+It is important to note that between the remote attestation and the key negotiation, a key binding mechanism between the  attestation result and the attester's identity must be established to prevent diversion attack. 
```
Suggested to add the definition of terms in the section notations as follows

    Key Binding:  A security protocol mechanism that cryptographically ties the results of a remote attestation to a specific communication session or entity identity. This ensures that the attestation evidence cannot be diverted and used by an attacker to impersonate the prover in a different context. 
    Diversion Attack:  A man-in-the-middle attack where an adversary intercepts and forwards attestation messages to trick a verifier into believing that the adversary's session is authenticated. The attacker's goal is to divert the authentication result and the subsequent privileged information (e.g., keys) to itself. Key Binding is the primary defense against this attack.

muhammad-usama-sardar replied:
@junzhanghw Could you share the source of these definitions?

junzhanghw replied:
@muhammad-usama-sardar , i summarize them by my own words as i do not find them in the attested-tls slides, if you can find the official definition and cite the source it will be better.

junzhanghw commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -110,43 +113,41 @@ Furthermore, the method of integrating key distribution into the remote attestat
+ 3. Evidence + ^    |
+    pubS +     |    |5. Attestation
+    Attester's |    | Result + pubS + Attester's identity
+    identity   |    v     1. Attestation
```
Is here pubS not equal to Attester's id?

muhammad-usama-sardar replied:
No, `identity` is different from `pubS` (= public key)

FrankXL replied:
The two are distinct: pubS serves as the material for key negotiation, which is a requirement of the business. The attester's ID plays a role in binding the evidence to the attester's entity identity during the remote attestation process, acting as a security mechanism to prevent diversion attacks.

junzhanghw replied:
So it seems that including attester id is a general approach to prevent diversion attack (even in the case that key negotiation is not needed).  Do we need to include it explicitly in the information flow, or we can simply mention it in the security consideration section? I prefer to decouple the part to handle diversion attack with the key negotiation inside attestation (which is the main focus of this draft).

FrankXL replied:
> So it seems that including attester id is a general approach to prevent diversion attack (even in the case that key negotiation is not needed). Do we need to include it explicitly in the information flow, or we can simply mention it in the security consideration section? I prefer to decouple the part to handle diversion attack with the key negotiation inside attestation (which is the main focus of this draft).

No, the binding of the attester ID is not only for preventing diversion attacks; the public key in the ID is also used for key distribution in use cases 1 and 3. Therefore, it needs to be explicitly included in the information flow. As for use case 2, the attester ID is indeed only used to prove the binding relationship with the attester result.

henkbirkholz commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -77,6 +77,9 @@ For example, to protect model parameters from leakage and tampering, model weigh
 At this point, there are multiple options for the secure channel protocol that can be established, such as TLS, IPSec, QUIC, OHTTP, etc.
 The user may only need to complete E2E key negotiation based on remote attestation.
 As for which secure protocol or application layer encryption the negotiated key is used for, and how it is used, there can be various implementation methods.
+* Scenario 3: A company intends to deploy its private large model or file system to a public cloud platform. However, for security reasons, it first uploads the encrypted objects to the cloud platform. Subsequently, it needs to conduct a security check on the TEE (Trusted Execution Environment) within the cloud platform's computing environment using remote attestation. Once the security check is  ...
```
```suggestion
* Scenario 3: A company intends to deploy its private large model or file system in a public cloud platform's trusted execution environment (TEE). To keep it deployed content confidential, the company first uploads the encrypted objects to the cloud platform. Subsequently, it conducts a security check on the cloud platform's TEE using remote attestation. Once the security check has passed, the decryption key is sent to decrypt the uploaded objects inside the TEE.
```

henkbirkholz replied:
These changes are just reshuffling a bit of the content. But! The text "within the cloud platform's computing environment" is now gone. Was that meant to be different from the TEE? For example, this could also have been the hosting environment in a CSP node, but I could not tell.

FrankXL replied:
Yes, TEE is sufficient and abstract enough, as this scenario is indeed not limited to public clouds.

henkbirkholz commented on draft-xia-rats-key-negotiation-integration.md:
```diff
@@ -77,6 +77,9 @@ For example, to protect model parameters from leakage and tampering, model weigh
 As for which secure protocol or application layer encryption the negotiated key is used for, and how it is used, there can be various implementation methods.
+* Scenario 3: A company intends to deploy its private large model or file system to a public cloud platform. However, for security reasons, it first uploads the encrypted objects to the cloud platform. Subsequently, it needs to conduct a security check on the TEE (Trusted Execution Environment) within the cloud platform's computing environment using remote attestation. Once the security check is  ...
+
+It is important to note that between the remote attestation and the key negotiation, a key binding mechanism between the  attestation result and the attester's identity must be established to prevent diversion attack.
```
```suggestion
It is important to note that between the remote attestation and the key negotiation, a key binding mechanism between the Attestation Result and the Attester's identity (e.g., an Attestation Key Certificate) has to be established to prevent diversion attack.
```

henkbirkholz replied:
Does this suggested change retain the intended meaning?

junzhanghw replied:
For me, it is not clear that, whether putting the attester id into the evidence (or put it into the meta data of evidence), and that of attestation result is already counted as key binding, or another key binding mechanism should be applied.

FrankXL replied:
> For me, it is not clear that, whether putting the attester id into the evidence (or put it into the meta data of evidence), and that of attestation result is already counted as key binding, or another key binding mechanism should be applied.


My understanding is that it would be more appropriate to place it in the metadata. The previous RATS AR probably did not specifically consider the issue of key binding.

FrankXL committed:
Update draft-xia-rats-key-negotiation-integration.md

accept.

Co-authored-by: Henk Birkholz <henkbirkholz@users.noreply.github.com>

FrankXL committed:
Update draft-xia-rats-key-negotiation-integration.md

ok

Co-authored-by: Henk Birkholz <henkbirkholz@users.noreply.github.com>

FrankXL committed:
Update draft-xia-rats-key-negotiation-integration.md

Based on Usama's suggestion, a new chapter (Implementation Status) has been added, which includes an introduction to the Trustee and descriptions of its relationship with use cases 1 and 3 in this document.

FrankXL committed:
Update draft-xia-rats-key-negotiation-integration.md

move key usage section under the Introduction charter.
move the diversion attack mitigation issue into the security consideration section.
Some Nits.

muhammad-usama-sardar commented:
Did you close it by mistake?
